packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts

```diff
@@ -312,6 +312,68 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
     }
   }
 
+  /**
+   * Enable connection logging for this load balancer.
+   *
+   * A region must be specified on the stack containing the load balancer; you cannot enable logging on
+   * environment-agnostic stacks.
+   *
+   * @see https://docs.aws.amazon.com/cdk/latest/guide/environments.html
+   */
+  public logConnectionLogs(bucket: s3.IBucket, prefix?: string) {
+    /**
+    * KMS key encryption is not supported on Connection Log bucket for ALB, the bucket must use Amazon S3-managed keys (SSE-S3).
+    * See https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-connection-logging.html#bucket-permissions-troubleshooting-connection
+    */
+    if (bucket.encryptionKey) {
+      throw new Error('Encryption key detected. Bucket encryption using KMS keys is unsupported');
+    }
+
+    prefix = prefix || '';
+    this.setAttribute('connection_logs.s3.enabled', 'true');
+    this.setAttribute('connection_logs.s3.bucket', bucket.bucketName.toString());
+    this.setAttribute('connection_logs.s3.prefix', prefix);
+
+    // https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-connection-logging.html
+    const logsDeliveryServicePrincipal = new ServicePrincipal('delivery.logs.amazonaws.com');
+    bucket.addToResourcePolicy(new PolicyStatement({
+      actions: ['s3:PutObject'],
+      principals: [this.resourcePolicyPrincipal()],
+      resources: [
+        bucket.arnForObjects(`${prefix ? prefix + '/' : ''}AWSLogs/${Stack.of(this).account}/*`),
+      ],
+    }));
+    // We still need this policy for the bucket using the ACL
+    bucket.addToResourcePolicy(
+      new PolicyStatement({
+        actions: ['s3:PutObject'],
+        principals: [logsDeliveryServicePrincipal],
+        resources: [
+          bucket.arnForObjects(`${prefix ? prefix + '/' : ''}AWSLogs/${this.env.account}/*`),
```

Comment:
for my own understanding, is there a difference in setting this value to `this.env.account` instead of `Stack.of(this).account` and purpose of using this instead of how we used it in line 343.

Reply:
Replaced `this.env.account` with `Stack.of(this).account`.
Since there is no `logConnectionLogs` method in `IApplicationLoadBalancer`, in the `logConnectionLogs` method, `Stack.of(this).account` and `this.env.account` will point to the same region.
So either one is fine, but `Stack.of(this).account` is clearer.